src/mpid/ch4/shm/posix/posix_coll_release_gather.h

```diff
@@ -150,7 +157,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_bcast_release_gather(void *buffer,
 
   fn_exit:
     MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_POSIX_MPI_BCAST_RELEASE_GATHER);
-    return mpi_errno;
+    return mpi_errno_ret;
```

Comment:
Wouldn't this get certain error return lost? For example, line 49: `mpi_errno = MPIR_Bcast_impl`, it will return `MPI_SUCCESS` no matter what error it gets.

Reply:
Maybe we should create a custom mpi_errno if (mpi_errno_ret>0)?

Or maybe I simply don't understand how `MPIR_ERR_ADD` works.

Reply:
Errors are not getting lost. That is the whole point of doing `ERR_ADD` instead of `ERR_POP`. I don't fully understand how it internally works. But this how other Bcast algorithms are propagating the error too.

Reply:
I was referring to line 49, where it calls fall-back `MPIR_Bcast_impl` and `goto fn_exit`. Maybe we should add a `MPIR_ERR_ADD` there before `goto fn_exit`?

Reply:
Oh. That is not an error.

Reply:
What happens when `MPIR_Bcast_impl(...)` returns an error?

Reply:
Sorry. I see  it now. I fixed it.